tests/core/test_variable.py

```diff
@@ -0,0 +1,548 @@
+    column = IntCol
+    entity = Person
+    definition_period = MONTH
+    label = u"Variable with stop attribute only coming before formula start."
+    end = '1990-01-01'
+
+    def formula_2000_01_01(self, individu, period):
+        return vectorize(self)
+
+
+def test_add__stop_attribute_before__one_formula__start():
+    check_error_at_add_variable(tax_benefit_system, stop_attribute_before__one_formula__start, 'You declared that "stop_attribute_before__one_formula__start" ends on "1990-01-01", but you wrote a formula to calculate it from "2000-01-01"')
+
+
+# end, formula with dates intervals overlap.
+
+class end_attribute_restrictive__one_formula(Variable):
+    column = IntCol
+    entity = Person
+    definition_period = MONTH
+    label = u"Variable with end attribute, one dated formula and dates intervals overlap."
+    end = '2001-01-01'
+
+    def formula_2001_01_01(self, individu, period):
+        return vectorize(self, 100)
+
+
+tax_benefit_system.add_variable(end_attribute_restrictive__one_formula)
+
+
+def test_call__end_attribute_restrictive__one_formula():
+    month = '2000-12'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute_restrictive__one_formula', month) == IntCol.default
+
+    month = '2001-01'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute_restrictive__one_formula', month) == 100
+
+    month = '2000-05'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute_restrictive__one_formula', month) == IntCol.default
+
+
+# formulas of different names (without dates overlap on formulas)
+
+class end_attribute__formulas__different_names(Variable):
+    column = IntCol
+    entity = Person
+    definition_period = MONTH
+    label = u"Variable with end attribute, multiple dated formulas with different names."
+    end = '2010-12-31'
+
+    def formula_2000_01_01(self, individu, period):
+        return vectorize(self, 100)
+
+    def formula_2005_01_01(self, individu, period):
+        return vectorize(self, 200)
+
+    def formula_2010_01_01(self, individu, period):
+        return vectorize(self, 300)
+
+
+tax_benefit_system.add_variable(end_attribute__formulas__different_names)
+
+
+def test_call__end_attribute__formulas__different_names():
+    month = '2000-01'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute__formulas__different_names', month) == 100
+
+    month = '2005-01'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute__formulas__different_names', month) == 200
+
+    month = '2010-12'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute__formulas__different_names', month) == 300
+
+
+def test_clone__end_attribute__formulas__different_names():
```

Comment:
I do not think it is wise to add system tests to OpenFisca in its current state, especially in a file that is already over 500 lines. Please don't rigidify stuff that nobody understands properly. It is welcome to add guarantees to new behaviour, but not very useful to check that we don't change a behaviour that might be flawed.